From a3eaf22e96063546ecf5734e87ac688c20044c11 Mon Sep 17 00:00:00 2001
From: David Morel <david.morel@vates.tech>
Date: Fri, 21 Jun 2024 11:46:24 +0200
Subject: [PATCH] ovs-ctl: set default ssl ciphers and protocols

Avoid weak ciphers using the command line options available in this
version of openvswitch (2.17). On older version (2.5) we used an openssl
configuration file to do so, but this version overrides the
settings in the configuration, so we have to use the command line
parameters --ssl-ciphers and --ssl-protocols.

Signed-off-by: David Morel <david.morel@vates.tech>
---
 utilities/ovs-ctl.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utilities/ovs-ctl.in b/utilities/ovs-ctl.in
index e6e07f4..a500e70 100644
--- a/utilities/ovs-ctl.in
+++ b/utilities/ovs-ctl.in
@@ -340,7 +340,7 @@ set_defaults () {
     OVS_VSWITCHD_PRIORITY=-10
     OVSDB_SERVER_WRAPPER=
     OVS_VSWITCHD_WRAPPER=
-    OVSDB_SERVER_OPTIONS=
+    OVSDB_SERVER_OPTIONS="--ssl-ciphers=AES256-GCM-SHA38:AES256-SHA256:AES256-SHA:AES128-GCM-SHA256:AES128-SHA256:AES128-SHA --ssl-protocols=TLSv1.2"
     OVS_VSWITCHD_OPTIONS=
 
     DB_FILE=$dbdir/conf.db
-- 
2.45.1

